diff --git a/com_google_riegeli/riegeli/records/BUILD b/com_google_riegeli/riegeli/records/BUILD
index 4c344f7..e208dc6 100644
--- a/com_google_riegeli/riegeli/records/BUILD
+++ b/com_google_riegeli/riegeli/records/BUILD
@@ -12,6 +12,7 @@ cc_library(
     srcs = ["record_reader.cc"],
     hdrs = ["record_reader.h"],
     deps = [
+        "@dbghelp",
         ":chunk_reader",
         ":record_position",
         ":records_metadata_cc_proto",
@@ -45,6 +46,7 @@ cc_library(
     srcs = ["record_writer.cc"],
     hdrs = ["record_writer.h"],
     deps = [
+        "@dbghelp",
         ":chunk_writer",
         ":record_position",
         ":records_metadata_cc_proto",
