diff --git a/com_google_riegeli/riegeli/bytes/fd_writer.h b/com_google_riegeli/riegeli/bytes/fd_writer.h
index ada2bc1..533ad99 100644
--- a/com_google_riegeli/riegeli/bytes/fd_writer.h
+++ b/com_google_riegeli/riegeli/bytes/fd_writer.h
@@ -158,7 +158,7 @@ class FdWriterBase : public BufferedWriter {
 
    private:
     absl::optional<std::string> assumed_filename_;
-    int mode_ = O_WRONLY | O_CREAT | O_TRUNC;
+    int mode_ = O_WRONLY | O_CREAT | O_TRUNC | O_BINARY;
     mode_t permissions_ = 0666;
     absl::optional<Position> assumed_pos_;
     absl::optional<Position> independent_pos_;
