diff --git a/envlogger/backends/tfds_backend_testlib.py b/envlogger/backends/tfds_backend_testlib.py
index c3b4df1..18f99d6 100644
--- a/envlogger/backends/tfds_backend_testlib.py
+++ b/envlogger/backends/tfds_backend_testlib.py
@@ -76,7 +76,7 @@ def catch_env_tfds_config(
       observation_info=tfds.features.Tensor(
           shape=(10, 5), dtype=tf.float32,
           encoding=tfds.features.Encoding.ZLIB),
-      action_info=tf.int64,
+      action_info=tf.int32,
       reward_info=tf.float64,
       discount_info=tf.float64,
       step_metadata_info={'timestamp': tf.int64},
